doc: Add Azure DevOps OAuth provider docs (#48822)
sourcegraph/sourcegraph: doc/admin/external_service/azuredevops.md

## Webhooks

Please consult [this page](../config/webhooks.md) in order to configure webhooks.

## Permissions syncing

[User-level permissions](../permissions/syncing.md#permission-syncing) syncing is supported for Azure DevOps code host connections. Here is the list of prerequisites:

1. Configure Azure DevOps as an OAuth provider by consulting [this page](../config/authorization_and_authentication.md#azure-devops-services)
2. Next verify that users can now sign up / login to your Sourcegraph instance with your Azure DevOps OAuth provider

> 3. Set the following to your Azure DevOps code host configuration:

Reviewer: ```suggestion
3. Set the following in your Azure DevOps code host connection:
```

## Webhooks

Please consult [this page](../config/webhooks.md) in order to configure webhooks.

## Permissions syncing

> User level permissions syncing is supported for Azure DevOps code host connections. Here is the list of prerequisites:

Reviewer: ```suggestion
User-level permissions syncing is supported for Azure DevOps code host connections. Here is the list of prerequisites:
```

I'd also link to the permissions page.

1. Configure Azure DevOps as an OAuth provider by consulting [this page](../config/authorization_and_authentication.md#azure-devops-services)
2. Next verify that users can now sign up / login to your Sourcegraph instance with your Azure DevOps OAuth provider
3. Set the following to your Azure DevOps code host configuration:

```
  "enforcePermissions": true
```

> 

Reviewer: Does this render correctly? Seems like the codeblock would need to be intended more?

Author: <img width="1262" alt="image" src="https://user-images.githubusercontent.com/2682729/223461333-1df906f2-1d90-45b7-93c8-81efffc07946.png">

This is how it renders at the moment. I didn't get why the codeblock needs to indent more though? 🤔

Author: Added the `json` tag and now it looks like this:

<img width="929" alt="image" src="https://user-images.githubusercontent.com/2682729/223462316-1ac50215-aba0-4a09-86b1-aba172b454df.png">

Reviewer: Before:

<img width="874" alt="screenshot_2023-03-07_17 44 46@2x" src="https://user-images.githubusercontent.com/1185253/223490216-d8ec9859-f4af-42f0-a811-458e0b0e968f.png">

After:

<img width="877" alt="screenshot_2023-03-07_17 45 14@2x" src="https://user-images.githubusercontent.com/1185253/223490329-9baac79a-dd8a-4c5b-a52a-37df5bba425b.png">

Look at the background of the code block and the margins of each item in the list.

---

Like this:

```
1. Configure Azure DevOps as an OAuth provider by consulting [this page](../config/authorization_and_authentication.md#azure-devops-services)
2. Next verify that users can now sign up / login to your Sourcegraph instance with your Azure DevOps OAuth provider
3. Set the following to your Azure DevOps code host configuration:

    ```json
      "enforcePermissions": true
    ```
4. For each Azure DevOps organization that is part of the `orgs` or `projects` list in the code host configuration, enable `Third-party application access via OAuth` from `Organization settings` -> `Security` -> `Policies`
```

See this doc for details: https://gist.github.com/clintel/1155906

(Or maybe there's a GitHub docs page now)

Reviewer: Oh and if we make it valid JSON then it renders nicely:

<img width="661" alt="screenshot_2023-03-07_17 47 23@2x" src="https://user-images.githubusercontent.com/1185253/223490897-c5ba69ec-81a8-4d48-a36b-4ad383450968.png">

```
3. Set the following to your Azure DevOps code host configuration:

    ```json
    {
      // [...]
      "enforcePermissions": true
    }
    ```
```
